### PR TITLE
Provide links in commit summaries in commits table/view list

### PR DESCRIPTION
**Pull request by gary-kim**
The commit summary and view lists now will provide links to referenced issues.

### CODE REVIEW

**Comment by lunny** [comment]
Could you give some screenshots before and after?

**Comment by lafriks** [comment]
This feature was intentionally removed in #6842

**Comment by gary-kim** [comment]
> This feature was intentionally removed in #6842

Whoops, did not notice that. What ui pages were effected by that bug? I'll see if I can address those inconsistencies.

**Comment by lafriks** [comment]
See issue #6809

**Comment by gary-kim** [comment]
So the closed issue was everything? Just to clarify, the feature was removed in case the entire commit message seemed like a hash which could lead to Gitea making it a commit link?
It seems like rather then addressing the bug, the change had removed the situations in which the bug can arise, which is a valid way to resolve issues but comes at the cost of removing a useful feature.

There is still the issue of what happens if the summary of a commit is a valid hash for another commit or referring to an issue. GitHub, for example, does not seem to have handled that situation and will link to the referred commit or issue rather the the commit whose summary contains the hash. This is only really an issue in places such as the files list and commit summaries on issues since they have no alternate link to the commit other then the commit summary. It can be made clear to the user that the summary is not a link to the actual commit using less but it would mean there is no link to the original commit. I'm not sure what we should do in that scenario. Anyone have any ideas?

For now, I'll look at making the `RenderCommitMessageLink` function check if hashs are valid commits first.

**Comment by gary-kim** [comment]
Blocked by #7713

**Comment by gary-kim** [comment]
@lafriks Now that #7713 is merged, do you think it would make sense to add this feature back in since the chance of an unintentional link is much lower?

**Comment by lunny** [comment]
Please rebase

**Comment by gary-kim** [comment]
@lunny Done

**Comment by gary-kim** [comment]
The branch is now a change to reenable message linking. It is now possible to do this without the issues reported in #6809 now that #7713 has made it so that possible commit hashes are checked before being turned into hashes.

@mrsdizzie Do you think it would be okay to add this back in now?

**Review by lafriks — APPROVED** [review]

**Comment by gary-kim** [comment]
@lafriks I also just added the commit message links to the files list. Do you want to recheck that?

**Review by guillep2k — APPROVED** [review]

File: templates/repo/view_list.tmpl [hunk 1]
```diff
@@ -83,7 +83,7 @@
 				{{end}}
 				<td class="message">
 					<span class="truncate has-emoji">
-						<a href="{{$.RepoLink}}/commit/{{$commit.ID}}" title="{{$commit.Summary}}">{{$commit.Summary}}</a>
+						{{RenderCommitMessageLink $commit.Message $.RepoLink (Printf "%s/commit/%s" $.RepoLink $commit.ID)  $.Repository.ComposeMetas}}
```
Comment (lafriks) [inline templates/repo/view_list.tmpl:86]: I don't think this is good idea to format this as there is not much point in this as links within links will add more confusion

Comment (gary-kim) [reply templates/repo/view_list.tmpl:86]: Maybe what we need is different styling on the links inside links to make it more clear to users.

Comment (lafriks) [reply templates/repo/view_list.tmpl:86]: If I remember correctly there was problems with links within links already in previous versions

Comment (lafriks) [reply templates/repo/view_list.tmpl:86]: imho this line change should be currently reverted until we come up with good solution for this

Comment (gary-kim) [reply templates/repo/view_list.tmpl:86]: What were the issues? I'd like to take a look as to whether they can be addressed.

EDIT: Whoops, your message didn't get updated when this comment was originally posted. Should I open a PR to revert that commit at least until all those issues you refer to have been ironed out? It is probably a good idea to address the issues in another branch but I'm not quite sure what the scope of the problems involving having links within links are yet.

Comment (lafriks) [reply templates/repo/view_list.tmpl:86]: First of all you can really tell apart where inner links begin and outer ends.
Need to check also if overflow hiding and elipsis adding does not breaks everyting if inner link happens to be at the end.

Yes, please open PR

Comment (gary-kim) [reply templates/repo/view_list.tmpl:86]: @lafriks #7923 

I think we can also use the PR to get some info about what the issues with the feature were.

**Pull request merged**